networkx/algorithms/min_edge_cover.py

```diff
@@ -0,0 +1,63 @@
+#TODO: Add Test Cases
+"""
+An edge cover of a graph is a set of edges such that every vertex of
+the graph is incident to at least one edge of the set.
+A minimum edge cover is an edge covering of smallest possible size.
+
+https://en.wikipedia.org/wiki/Edge_cover
+
+"""
+import networkx as nx
+__all__ = ["min_edge_cover"]
+__author__ = """Nishant Nikhil (nishantiam@gmail.com)"""
+
+def min_edge_cover(G):
+
+    """
+    A smallest edge cover can be found in polynomial time by finding
+    a maximum matching and extending it greedily so that all vertices
+    are covered.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        Undirected graph
+        Unweighted graph
+
+    Returns
+    -------
+    min_cover : dictionary
+        The matching is returned as a dictionary, min_weighted_edge_cover,
+        such that min_weighted_edge_cover[v] has a list of nodes so that
+        the edge joining them occurs in minimum edge cover.
+        If no edge cover exists, return an empty list.
+
+    Notes
+    -----
+    Worst Case runtime of maximum_matching is O(number_of_nodes ** 3)
+    And the greedy extension works in O(number_of_nodes)
+    So run time of the function is O(number_of_nodes ** 3)
+    """
+    maximum_matching = nx.max_weight_matching(G, maxcardinality=True)
+    min_cover = {}
+    #min_weighted_edge_cover is superset of maximum_matching
```

Comment:
A space after `#`. You can check for pep8 issues by installing `pep8` and running it on the file.